foundation-ui/auth/AuthService.js

```diff
@@ -0,0 +1,105 @@
+import Hooks from '../hooks';
+
+const hooks = Hooks();
+
+/**
+ * AuthService relies on a single instance of Hooks.
+ * It utilses the hooks actions to notify AuthPoints of changes and
+ * filters to allow external packages to register for and change the content
+ * output in the AuthPoint
+ *
+ * @example
+ * User of service would listen for updates and possibly change it
+ * import {AuthService} from 'foundation-ui/mount';
+ * AuthService.register('myID', function (value) {
+ *   return value.replace('my', 'myChanged');
+ * });
+ *
+ * AuthPoint would register itself with the AuthService to be able to get
+ * updates on content
+ * import {AuthService} from 'foundation-ui/mount';
+ * function updateContent() {
+ *   let newContent = AuthService.getContent(
+ *     'myID',
+ *     React.Children.toArray(children),
+ *     props
+ *   );
+ *
+ *   // Do something with newContent
+ * }
+ * AuthService.addListener('myID', updateContent);
+ */
+const AuthService = {
+  // Package API
+  /**
+   * Registers package for updates on this AuthPoint permission
+   * @param  {String} permission of MointPoint to register package for
+   * @param  {Function} callback to call when content is being fetched
+   * @param  {Number} priority to register package callback with
+   * [-Infinity; Infinity]
+   */
+  register(permission, callback, priority) {
```

Comment:
I like us to provide a stricter api by defining and testing against an `AuthenticationAdapter` type. The adapter is then used to verify that a user has the required permission.

I picture the authentication service to work like illustrated (!) in the following "example":

```
class AuthenticationService {

    constructor(){
        this.authenticationAdapter = [];
    }

    /**
     * Register an adapter to perform authentication and authorization 
     * checks
     * @param {AuthenticationAdapter} adapter
     */
    registerAuthenticationAdapter(adapter) {
        if(!(adapter instanceof AuthenticationAdapter)){
            throw new TypeError('Adapter should be an instance of AuthenticationAdapter');
        }

        this.authenticationAdapter.push(adapter);
    }

    /**
    * Check  authentication adapters to verify that the user has 
    * the required permission
    */
    isAuthorized(permission){
        return  this.authenticationAdapter.every(function(adapter){
            return adapater.isAuthorized(permission) 
        });
    }
}
```

This API will give enough flexibility to deal with different authentication backends (e.g. oAuth) while making sure that the provided API is simple and easy to use.